### PR TITLE
Highlight diagram if already open

### PR DESCRIPTION
**Pull request by EstherDarkish**
Highlight a diagram's header when trying to open it again if it is already open.

### CODE REVIEW

File: src/components/diagrams/diagram.js [hunk 1]
```diff
@@ -58,6 +59,28 @@ const Diagram = (props) => {
...
     );
 
+    /**
+     * BLINKING HEADER MONITORING
+     */
+
+    const diagramStates = useSelector((state) => state.diagramStates);
+    const headerRef = useRef();
+
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/diagrams/diagram.js]: Yes, that's better that way. I updated it.
I also changed the useSelector lamda expression to decrease the number of renders needed for the effect.

File: src/components/diagrams/diagram-header.js [hunk 2]
```diff
@@ -52,8 +77,42 @@ const DiagramHeader = (props) => {
...
+        if (needsToBlink) {
+            dispatch(stopDiagramBlink());
+            if (!blinking) {
+                setBlinking(true);
+                setTimeout(() => {
+                    setBlinking(false);
+                }, BLINK_LENGTH_MS);
+            }
+        }
+    }, [needsToBlink, dispatch, props.svgType, props.diagramId, blinking]);
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/diagrams/diagram-header.js:104]: I like to have my props explicitly defined

File: src/redux/reducer.js [hunk 3]
```diff
@@ -812,6 +814,15 @@ export const reducer = createReducer(initialState, {
             (diagram) => !idsToClose.has(diagram.id)
         );
     },
+    [STOP_DIAGRAM_BLINK]: (state) => {
+        let diagramStates = state.diagramStates;
```
Comment (flomillot) [inline src/redux/reducer.js:818]: Why this copy ?

File: src/components/diagrams/diagram-header.js [hunk 4]
```diff
@@ -14,6 +15,10 @@ import PushPinOutlinedIcon from '@mui/icons-material/PushPinOutlined';
 import CloseIcon from '@mui/icons-material/Close';
 import PropTypes from 'prop-types';
 import makeStyles from '@mui/styles/makeStyles';
+import clsx from 'clsx';
+import { stopDiagramBlink } from '../../redux/actions';
+
+const BLINK_LENGTH_MS = 3000;
```
Comment (flomillot) [inline src/components/diagrams/diagram-header.js:21]: It's slightly slow for me, maybe 2 sec with '0%, 20%' ? I don't know what think the PO about it, it's their choice at the end

File: src/components/diagrams/diagram-header.js [hunk 5]
```diff
@@ -22,6 +27,11 @@ const useStyles = makeStyles((theme) => ({
         flexDirection: 'row',
         wordBreak: 'break-all',
         backgroundColor: theme.palette.background.default,
+        borderBottom: 'solid 1px',
+        borderBottomColor:
+            theme.palette.mode === 'light'
+                ? theme.palette.action.selected
+                : 'rgba(0,0,0,0)',
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/diagrams/diagram-header.js]: Thanks, didn't knew this one.

File: src/redux/reducer.js (hunk 3 above)
Comment (EstherDarkish) [reply src/redux/reducer.js]: Removed

File: src/components/diagrams/diagram-header.js (hunk 4 above)
Comment (EstherDarkish) [reply src/components/diagrams/diagram-header.js]: Ok, I slightly changed it, now 1800ms and 0-25%

**Review by flomillot — APPROVED** [review]

**Review by Walid-Sahnoun — APPROVED** [review]

**Pull request merged**